app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +69,4106 @@ const PRE_1559_BLOCK = {
+        await withController(async ({ controller, network }) => {
+          network.mockEssentialRpcCalls();
+
+          const promiseForInfuraIsUnblocked = waitForEvent({
+            controller,
+            eventName: 'infuraIsUnblocked',
+            operation: async () => {
+              await controller.lookupNetwork();
+            },
+          });
+
+          await expect(promiseForInfuraIsUnblocked).toNeverResolve();
+        });
+      });
+
+      it('does not emit infuraIsBlocked', async () => {
+        await withController(async ({ controller, network }) => {
+          network.mockEssentialRpcCalls();
+
+          const promiseForInfuraIsBlocked = waitForEvent({
+            controller,
+            eventName: 'infuraIsBlocked',
+            operation: async () => {
+              await controller.lookupNetwork();
+            },
+          });
+
+          await expect(promiseForInfuraIsBlocked).toNeverResolve();
+        });
+      });
+    });
+
+    for (const {
+      networkName,
+      networkType,
+      networkVersion,
+    } of INFURA_NETWORKS) {
+      describe(`when the type in the provider configuration is "${networkType}"`, () => {
+        describe('if the request for eth_blockNumber responds successfully', () => {
+          it('emits infuraIsUnblocked as long as the network has not changed by the time the request ends', async () => {
+            await withController(
+              {
+                state: {
+                  provider: {
+                    type: networkType,
+                  },
+                },
+              },
+              async ({ controller, network }) => {
+                network.mockEssentialRpcCalls({
+                  eth_blockNumber: {
+                    response: {
+                      result: '0x42',
+                    },
+                  },
+                });
+                await withoutCallingLookupNetwork({
+                  controller,
+                  operation: async () => {
+                    await controller.initializeProvider();
+                  },
+                });
+
+                const infuraIsUnblocked = await waitForEvent({
+                  controller,
+                  eventName: 'infuraIsUnblocked',
+                  operation: async () => {
+                    await controller.lookupNetwork();
+                  },
+                });
+
+                expect(infuraIsUnblocked).toBe(true);
+              },
+            );
+          });
+
+          it('does not emit infuraIsUnblocked if the network has changed by the time the request ends', async () => {
+            await withController(
+              {
+                state: {
+                  provider: {
+                    type: networkType,
+                  },
+                },
+              },
+              async ({ controller }) => {
+                const network1 = new NetworkCommunications({
```

Comment:
Could the `network` argument from `withController` be used here instead?

Reply:
Yeah, I wanted to make the initial network more explicit to differentiate it from `network2`, but it's inconsistent with how I do things in other tests, so I might as well be consistent. Changed in [8319bb5](https://github.com/MetaMask/metamask-extension/pull/17498/commits/8319bb51ebf5b14302f5b6cfd93d7b4b74d73426).